deployer/scripts/validate.sh

```diff
@@ -67,3 +63,100 @@ function check_routes() {
   [ "$failure" = false ] || return 1
 }
 
+function validate_deployment() {
+  printf '\n%s\n' 'DEPLOYMENT VALIDATION'
+  os::int::util::validate check_secrets check_deployer
+  wait_for_elasticsearch
+  os::int::util::validate check_elasticsearch
+}
+
+function check_secrets() {
+  local tmpl secret keys out
+  tmpl='{{range $k, $_ := .data}}{{$k}}{{"\n"}}{{end}}'
+  while read secret keys; do
+    out=$(oc get "secret/$secret" --template "$tmpl" | xargs echo)
+    [ "$out" == "$keys" ] && continue
+    validation_failure \
+      "Wrong data on deployer-generated secret $secret" \
+      "$(printf 'Expected:\n%s\nGot:\n%s\n' "$keys" "$out")"
+    return 1
+  done <<-'EOF'
+		apiman-elasticsearch \
+			ca.crt client.crt client.key client.keystore client.keystore.password \
+			keystore keystore.password searchguard-node-key truststore \
+			truststore.password
+		apiman-console \
+			ca.crt client.crt client.key client.keystore client.keystore.password \
+			keystore keystore.password truststore truststore.password
+		apiman-gateway \
+			ca.crt client.crt client.key client.keystore client.keystore.password \
+			keystore keystore.password truststore truststore.password
+		EOF
+}
+
+function check_deployer() {
+  local line out
+  while read line; do
+    out=$(os::int::util::check_exists $line 2>&1) && continue
+    printf >&2 '%s' "$out"
+    return 1
+  done <<-'EOF'
+		imagestreams apiman-console apiman-elasticsearch apiman-gateway
+		serviceaccounts \
+			apiman-console apiman-deployer apiman-elasticsearch apiman-gateway
+		deploymentconfigs apiman-console apiman-gateway
+		deploymentconfigs --selector component=elasticsearch
+		services apiman-console apiman-es-cluster apiman-gateway apiman-storage
+		routes apiman-console apiman-gateway
+		EOF
+}
+
+wait_for_elasticsearch() {
+  # curl(1):
+  # 28: Operation timeout.
+  # 52: The server didn't reply anything.
+  local deadline ret
+  deadline=$(($(date +%s) + 3 * 60))
+  while [ "$(date +%s)" -lt "$deadline" ]; do
```

Comment:
we should probably have our own "wait_for" function in common sort of like in origin's test scripts. it's probably a little awkward to do that in this environment though where you want to show the user something contextual.